### ISSUE TITLE
Bug: [vue/xstate-inspect] Impossible to inspect in iframe

### ISSUE THREAD

**Issue by SteamWind** [davidkpiano/xstate#3143]
### Description

As discussed in the [thread on the Stately discord](https://discord.com/channels/795785288994652170/948916730065993738/951105240114036796), it is impossible to use the iframe to inspect the state machine with vue if the iframe is in a component and the machines are initialized in that component (or elsewhere).

Indeed, the documentation says that you should run `inspect()` before any other code, for vue, the best being in `main.js`. But if you do this in this order, you can't see the inspector in the iframe (white iframe).

### Expected result

**(a)** When you run inspect before any code with the iframe option, the iframe displays the inspector.

OR

**(b)** When `inspect()` is run after the component containing the iframe has been initialized, the iframe detects and displays the previously run machines.

### Actual result

**(a)** When you run inspect before any code with the iframe option, the iframe displays nothing. (white iframe)
![image](https://user-images.githubusercontent.com/5443726/157670750-4b9391ba-c2a4-4abd-b701-a7299b2ba9f2.png)

OR

**(b)** When inspect is run after the component containing the iframe has been initialized, the inspector is launched but does not detect the machines that were previously launched. (Waiting for connection...)
![image](https://user-images.githubusercontent.com/5443726/157670947-3699fa2c-07a2-4bd7-9bac-5f0c0aa655e0.png)

### Reproduction
[sandbox **(a)**](https://codesandbox.io/s/bug-vue-xstate-inspect-impossible-to-inspect-in-iframe-a-f1s3l0)
[sandbox **(b)**](https://codesandbox.io/s/bug-vue-xstate-inspect-impossible-to-inspect-in-iframe-b-kuttbz?file=/src/main.js)

**Comment by itai-gendler** [comment]
were you able to solve this issue?
i'm also stuck on this trying to have an iframe in a compoenent.

**Comment by Andarist** [comment]
We are going to discuss changes to the app/inspector protocol~ to make use cases like this work out of the box. Stay tuned.

**Comment by davidkpiano** [comment]
The workaround right now is to add the iframe _outside_ of Vue:

```html
  <body>
    <noscript>
      <strong
        >We're sorry but codesandbox doesn't work properly without JavaScript
        enabled. Please enable it to continue.</strong
      >
    </noscript>
    <div id="app"></div>
    <iframe data-xstate></iframe>

    <!-- built files will be auto injected -->
  </body>
```

**Comment by SteamWind** [comment]
@itai-gendler 
> were you able to solve this issue? i'm also stuck on this trying to have an iframe in a compoenent.

No, I used the external window finally, but I had to do weird things to make it work. I had to destroy and reopen the window everytime we `inspect()`.

**Comment by SteamWind** [comment]
What did you came up with?

> We are going to discuss changes to the app/inspector protocol~

I saw news about stately sky. But it's not the same use case.

**Comment by davidkpiano** [comment]
Doing some issue cleanup; Stately Inspect should work with Vue projects normally, but if the issue still occurs, please open an issue here: https://github.com/statelyai/inspect/
